### PR TITLE
Updated 'fetch' & 'setup' commands to download dependencies

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: bin/impl/fetch.sh [hunk 1]
```diff
@@ -118,7 +118,7 @@ metrics)
   rm -rf "$GF_PATH"
   ;;
 *)
-  echo "Usage: uno fetch <dependency>"
+  echo "Usage: uno fetch <dependency1> [<dependency2>]"
```
Comment (keith-turner) [inline bin/impl/fetch.sh]: I looked for a standard shell syntax for one or more arguments.  It seems using ellipses is a standard way to do this.  That would look like `uno fetch <dependency>...` and would indicate 1 or more deps.  Below is what I found during my search.

 * In the process of searching I found [docopt](http://docopt.org/), which looks like a neat project. Docopt uses ellipses for one or more args.    
 * See 12.1.9 at http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html
 * After finding above I looked at help for tar and grep.  Both use `[FILE]...` which indicates zero or more files.

File: bin/uno [hunk 2]
```diff
@@ -97,15 +102,15 @@ version)
...
-  echo "  fetch <dependency>  Fetches dependencies by either building or downloading them (as configured by env.sh)"
-  echo "                      Run 'uno fetch all' to fetch all dependencies. Run 'uno fetch' for list of dependencies."
-  echo "  setup <command>     Sets up development cluster (clearing any existing data)"
-  echo "                      Run 'uno setup fluo' to setup all possible software. Run 'uno setup' for list of commands."
-  echo "  kill                Kills all software"
-  echo "  ashell              Runs the Accumulo shell"
-  echo "  env                 Prints out shell configuration for PATH and common environment variables."
-  echo "                      Add '--paths' or '--vars' command to limit what is printed."
-  echo "  version <dep>       Prints out configured version for dependency"
+  echo "  fetch <dep1> [<dep2>]  Fetches one or more dependencies by either building or downloading them (as configured by env.sh)"
```
Comment (keith-turner) [inline bin/uno]: this could also use `...`

File: bin/impl/fetch.sh (hunk 1 above)
Comment (mikewalch) [inline bin/impl/fetch.sh]: this is fixed now

File: bin/uno (hunk 2 above)
Comment (mikewalch) [inline bin/uno]: also fixed

**Comment by mikewalch** [comment]
I pushed another commit that also updated 'setup' command to accept multiple components

File: bin/uno [hunk 3]
```diff
@@ -62,30 +62,43 @@ fetch)
...
-    echo "    fluo       Sets up Fluo along with Accumulo (and its dependences)."
-    echo "               Also, sets up metrics server (if SETUP_METRICS=true in env.sh)"
-    echo "    fluo-only  Sets up Fluo only. Useful for upgrading Fluo without losing cluster data."
+  if [[ -z "${@:2}" ]]; then
+    echo "Usage: uno setup <component>..."
+    echo -e "\nPossible components:\n"
+    echo "    all        Sets up all of the following components"
+    echo "    accumulo   Sets up Accumulo along with Hadoop & Zookeeper"
+    echo "    spark      Sets up Spark"
+    echo "    fluo       Sets up Fluo. Requires Accumulo to be set up."
```
Comment (keith-turner) [inline bin/uno]: Why have accumulo setup its deps and fluo not setup its deps?

File: README.md [hunk 4]
```diff
@@ -102,29 +98,33 @@ The `setup` command will install the downloaded tarballs to the directory set by
...
 
-        uno setup fluo
+        uno setup accumulo fluo
 
 3. Sets up Apache Fluo only. This will stop any previously running Fluo applications but it will not
    wipe your cluster. If you want upgrade Fluo without wiping your cluster, run `uno fetch fluo`
    before running this command.
 
-        uno setup fluo-only
+        uno setup fluo
```
Comment (keith-turner) [inline README.md]: maybe setup could have a --no-dependencies option?   By default will setup deps, unless this option is specified.
```
uno setup accumulo --no-dependencies
```


```
uno setup fluo --no-dependencies
```

File: bin/uno (hunk 3 above)
Comment (mikewalch) [inline bin/uno]: There are cases where you want to setup fluo only.

File: README.md (hunk 4 above)
Comment (mikewalch) [inline README.md]: That sounds good to me.  However, I would like to shorten it to `--no-deps` and I think I will only do it for Fluo at this time.

Comment (mikewalch) [inline README.md]: I updated to the PR.  I was about to add it to both Fluo & Accumulo

File: README.md [hunk 5]
```diff
@@ -36,10 +36,9 @@ have satisfied the requirements mentioned above.
 ```bash
 git clone https://github.com/astralway/uno.git
 cd uno
-./bin/uno fetch all
-./bin/uno setup fluo
-# bash specific command to add Fluo, Accumulo, Hadoop, etc to path of current shell
-eval "$(./bin/uno env)"
+./bin/uno fetch fluo                # Fetches binary tarballs of Fluo and its dependencies
```
Comment (keith-turner) [inline README.md]: the added comments are nice

File: bin/uno [hunk 6]
```diff
@@ -45,42 +45,66 @@ case "$1" in
...
     echo "    metrics    Sets up metrics service (InfluxDB + Grafana)."
+    echo "Options:"
+    echo "    --no-deps  Dependencies will be setup unless this option is specified. Only works for fluo & accumulo components."
     exit 1
-  esac
-	;;
+  fi
+
+  setup_opts=""
+  if [[ "${@:2}" == *"--no-deps" ]]; then
```
Comment (keith-turner) [inline bin/uno]: if `uno setup fluo spark --no-deps` is run, what will this do?

File: bin/uno [hunk 7]
```diff
@@ -45,42 +45,66 @@ case "$1" in
...
+  fi
+
+  for component in "${@:2}"; do
+    case "$component" in
+      all)
+        "$bin"/impl/setup-fluo.sh
+        "$bin"/impl/setup-spark.sh
+        "$bin"/impl/setup-metrics.sh
+        ;;
+      accumulo)
```
Comment (keith-turner) [inline bin/uno]: if someone run `uno setup accumulo flo` will accumulo be setup twice?

File: bin/uno (hunk 6 above)
Comment (mikewalch) [inline bin/uno]: I changed setup to only accept one component.

File: bin/uno [hunk 8]
```diff
@@ -48,62 +48,46 @@ fetch)
...
-    esac
-  done
+  case "$2" in
+    all)
+      "$bin"/impl/setup-fluo.sh
+      "$bin"/impl/setup-spark.sh
+      "$bin"/impl/setup-metrics.sh
+      ;;
+    accumulo)
+      "$bin"/impl/setup-accumulo.sh $3
```
Comment (keith-turner) [inline bin/uno]: could be additional arguments after `$3`

File: bin/uno [hunk 9]
```diff
@@ -48,62 +48,46 @@ fetch)
...
+      ;;
+    fluo)
+      "$bin"/impl/setup-fluo.sh $3
+      ;;
+    spark)
+      "$bin"/impl/setup-spark.sh
```
Comment (keith-turner) [inline bin/uno]: could be additional args

File: bin/uno (hunk 8 above)
Comment (mikewalch) [inline bin/uno]: created #126

File: bin/uno (hunk 9 above)
Comment (mikewalch) [inline bin/uno]: created #126

**Pull request merged**